Combobox.js

```diff
@@ -394,89 +434,45 @@ define([
 							// before the dropdown closes.
 							this.closeDropDown(true/*refocus*/);
+
+							// Re-initialize the query.
+							if (this.getInitialQuery !== null && !this.hasDownArrow) {
+								this.list.query =
+									(typeof this.resetQuery === "function") ?
+										this.getInitialQuery() : this.getInitialQuery;
+							}
 						}.bind(this), 100); // worth exposing a property for the delay?
 					}
 				}
 			}.bind(this));
 
 			// React to interactive changes of selected items
 			this.list.on("selection-change", function () {
-				if (this.selectionMode === "single") {
-					this._validateSingle();
-				}
+				this._validateInput(true);
 				this.handleOnInput(this.value); // emit "input" event
 			}.bind(this));
 
-			// React to programmatic changes of selected items
-			this.list.observe(function (oldValues) {
-				if ("selectedItems" in oldValues) {
-					if (this.selectionMode === "single") {
-						this._validateSingle();
-						// do not emit "input" event for programmatic changes
-					} else if (this.selectionMode === "multiple") {
-						this._validateMultiple(this._popupInput || this.inputNode);
-					}
-				}
-			}.bind(this));
+			this.list.on("query-success", this._setSelectedItems.bind(this));
 
 			this._prepareInput(this.inputNode);
-		},
 
-		/**
-		 * Sets the initial value of the widget. If the widget is inside a form,
-		 * also called when reseting the form.
-		 * @private
-		 */
-		_initValue: function () {
-			if (this.selectionMode === "single") {
-				// Returns true if List content already available, false otherwise.
-				var initValueSingleMode = function () {
-					var selectedItem = this.list.selectedItem;
-					var done = false;
-					var value, label;
-					if (selectedItem) {
-						label = this._getItemLabel(selectedItem);
-						value = this._getItemValue(selectedItem);
-						done = true;
-					} else {
-						var firstItemRenderer = this.list.getItemRendererByIndex(0);
-						if (firstItemRenderer) {
-							label = this._getItemRendererLabel(firstItemRenderer);
-							value = this._getItemRendererValue(firstItemRenderer);
-							// Like the native select, the first item is selected.
-							this.list.selectedItem = firstItemRenderer.item;
-							done = true;
-						}
+			if (this._useCenteredDropDown()) {
+				this.on("click", function () {
+					if (!this.disabled) {
+						this.openDropDown(false);
 					}
-					if (done) {
-						this.inputNode.value = label;
-						// Initialize widget's value
-						this._set("value", value);
-						this.valueNode.value = value;
+				}.bind(this));
+			} else if (!this.minFilterChars) {
+				this.on("mousedown", function (evt) {
+					// event could be triggered by the down arrow. If so, we do not react to it.
+					if (!this.disabled && evt.srcElement !== this.buttonNode) {
+						if (!this.opened) {
+							this.openDropDown(false);
+						} else {
+							this.closeDropDown(true);
+						}
```

Comment:
I'm a bit worried about closing the dropdown due to a click on the `<input>` (as opposed to `this.buttonNode`), because the user might click the `<input>` just to move the caret.   What do you think?